### PR TITLE
Example: Radar component and demo

### PR DESCRIPTION
**Pull request by imudin**
Radar component is usually used to show where we are looking at in 360 image or VR viewers, inspired by krpano's radar plugin https://krpano.com/tours/bkeller/index.html

### CODE REVIEW

File: examples/components/Home.vue [hunk 1]
```diff
@@ -5,6 +5,7 @@
       <ul class="examples">
         <li><router-link to="/stars">Stars</router-link></li>
         <li><router-link to="/vuex-example">Vuex Example</router-link></li>
+        <li><router-link to="/wedge2-demo">Field Of Example</router-link></li>
```
Comment (Ldoppea) [inline examples/components/Home.vue]: Should be `/radar-demo` and `Radar Example`

File: examples/components/Radar.vue [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      this.handleDirectionChange(false);
+    }
+  },
+  methods: {
+    handleAngleChange(angle) {
+      this.config.rotation = this.config.direction - this.config.angle / 2.0
+    },
+    handleDirectionChange(angle) {
+      // todo need better solution
+      const shape = this.$parent.getStage().children[0]
```
Comment (Ldoppea) [inline examples/components/Radar.vue]: Then use `const shape = this.$refs.wedge.getStage()` so you don't have to care about parent component

File: examples/components/Radar.vue [hunk 3]
```diff
@@ -0,0 +1,43 @@
+<template>
+  <v-wedge :config='config'></v-wedge>
```
Comment (Ldoppea) [inline examples/components/Radar.vue]: You can add `ref="wedge"` so you can access it later from `methods`

File: examples/components/RadarDemo.vue [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+      fillRadialGradientEndPoint: 0,
+      fillRadialGradientEndRadius: 100,
+      fillRadialGradientColorStops: [0, 'red', 0.5, 'blue', 1, 'rgba(255,255,255,0)'],
+      opacity: 0.8,
+      stroke: 'transparent',
+      strokeWidth: 1,
+      scaleX: 1,
+      scaleY: 1,
+      startScale: 1,
+      handler: vm.wedgeHangler
```
Comment (Ldoppea) [inline examples/components/RadarDemo.vue]: Why is `handler` for?

File: examples/components/Radar.vue (hunk 2 above)
Comment (imudin) [reply examples/components/Radar.vue]: wow awesome, Thanks a lot! The reason why I did pull request is I wanted help :) and some correction of my code. I am new in vuejs

File: examples/components/Radar.vue (hunk 3 above)
Comment (imudin) [reply examples/components/Radar.vue]: I appreciate that you gave comments. I will update it

File: examples/components/RadarDemo.vue (hunk 4 above)
Comment (imudin) [reply examples/components/RadarDemo.vue]: I would like to add handler for dragging the radar so that it will rotate around its center. But I cannot do it now :(

File: examples/components/Home.vue (hunk 1 above)
Comment (imudin) [reply examples/components/Home.vue]: Yes right, I renamed wedge2 to radar, and forgot to update here

File: examples/components/RadarDemo.vue (hunk 4 above)
Comment (Ldoppea) [reply examples/components/RadarDemo.vue]: I'm not sure what "center" is for this shape or what is the difference between `direction` and `rotation` in your code.
But you can create a new shape as rotation handler (ie : `v-circle`) and use the `dragBoundFunc` property to constraint your shape movement using `atan2`, `cos` and `sin` helpers. Also in that function you can compute the new rotation of your wedge.

Also maybe you'll need to refactor your code a bit. `dragstart` and `dragend` are on top stage and this may constraint you in future gestures implementation inside this stage. You should move them directly on the radar component, or maybe inside of it (so the component is responsible for its own position).

File: examples/components/Radar.vue [hunk 5]
```diff
@@ -1,5 +1,18 @@
+/**
+  Radar is usually used to describe where we are looking at in 360 panorama viewers or VR projects,
+  inspired by krpano's radar plugin https://krpano.com/tours/bkeller/index.html
+  angle: field of view
+  direction: center angle of the view
+*/
+
 <template>
-  <v-wedge :config='config'></v-wedge>
+<div>
```
Comment (Ldoppea) [inline examples/components/Radar.vue]: Do not use `div` in Konva components, this is a DOM elements and here you are in a canvas context. Use `v-group` instead.

File: examples/components/RadarDemo.vue [hunk 6]
```diff
@@ -42,39 +39,6 @@ export default {
...
-      shape.to({
-        duration: 0.5,
-        easing: Konva.Easings.ElasticEaseOut,
-        scaleX: component.config.startScale,
-        scaleY: component.config.startScale,
-        shadowOffsetX: 5,
-        shadowOffsetY: 5
-      });
-    }
-  },
```
Comment (Ldoppea) [inline examples/components/RadarDemo.vue]: Ok so you started to refactor dragstart and dragend, forgot about my previous comment in previous commit 😄

File: examples/components/Radar.vue [hunk 7]
```diff
@@ -1,5 +1,18 @@
+/**
+  Radar is usually used to describe where we are looking at in 360 panorama viewers or VR projects,
+  inspired by krpano's radar plugin https://krpano.com/tours/bkeller/index.html
+  angle: field of view
+  direction: center angle of the view
+*/
+
```
Comment (Ldoppea) [inline examples/components/Radar.vue]: If I understand correctly, you don't need to drag&move you component along x and y positions, do you?
Just to be sure because actual demo allow drag&move and this may impact how you'll implement the rotation.

File: examples/components/Radar.vue [hunk 8]
```diff
@@ -29,7 +42,10 @@ export default {
     },
     handleDirectionChange(angle) {
       // todo need better solution
-      const shape = this.$parent.getStage().children[0]
+      // const shape = this.$parent.getStage().children[0]
+
+      // got better solution now
+      const shape = this.$refs.wedge.getStage()
 
       var tween = new Konva.Tween({
```
Comment (Ldoppea) [inline examples/components/Radar.vue]: Here you should make `tween` an instance variable and stop it every time before recreating a new one.

Also use `let` instead of `var` to avoid variable scope collisions.

File: examples/components/RadarDemo.vue [hunk 9]
```diff
@@ -84,21 +48,20 @@ export default {
       radius: 100,
       angle: 60,
       direction: 90,
-      draggable: true,
+      draggable: true,  // todo: after dragging, changing direction and angle makes radar to go back initial position
```
Comment (Ldoppea) [inline examples/components/RadarDemo.vue]: Regarding previous comment, just verify you really need it to be draggable before you fix the bug.

File: examples/components/Radar.vue (hunk 7 above)
Comment (imudin) [reply examples/components/Radar.vue]: I need to drag and move by dragging itself, and should be rotated by dragging its handler

File: examples/components/RadarDemo.vue (hunk 9 above)
Comment (imudin) [reply examples/components/RadarDemo.vue]: Yeah I really need it to be draggable and rotatable :)

File: examples/components/Radar.vue (hunk 5 above)
Comment (imudin) [reply examples/components/Radar.vue]: wow `v-group` is what I needed!

**Comment by rafaesc** [comment]
Awesome work @imudin @Ldoppea,

I saw some TODOs, say me if it already ready to merge it

Regards

File: examples/components/RadarDemo.vue [hunk 10]
```diff
@@ -45,28 +47,29 @@ export default {
...
       strokeWidth: 1,
       scaleX: 1,
       scaleY: 1,
-      startScale: 1,
-      handler: vm.wedgeHangler   // todo: draw handler for radar to rotate (to change direction)
+      startScale: 1
     }
-    vm.wedgeHangler = {
+
+    vm.handler = { // todo: draw handler for radar to rotate (to change direction)
```
Comment (Ldoppea) [inline examples/components/RadarDemo.vue]: Not a todo anymore (same for the other todo on wedge.draggable)

File: examples/components/Radar.vue [hunk 11]
```diff
@@ -6,17 +6,28 @@
...
+    @dragend='onHandlerDragEnd'>
+  </v-circle>   
+</v-group>
   
 </template>
 
 <script>
 import {Werge} from 'konva'
+
+let tween;
```
Comment (Ldoppea) [inline examples/components/Radar.vue]: `tween` seems not to be used anymore.

File: examples/components/RadarDemo.vue [hunk 12]
```diff
@@ -45,28 +47,29 @@ export default {
...
       radius: 10,
-      angle: 60,
-      opacity: 0.8,
-      strokeWidth: 1
+      strokeWidth: 2,
+      stroke: 'rgba(0,0,255,0.5)',
+      fill: 'rgba(255,255,255,1)',
+      draggable: true // this should be always true by default
```
Comment (Ldoppea) [inline examples/components/RadarDemo.vue]: Because you want to change opacity with `onHandlerDragStart()` and `onHandlerDragEnd()`, then you should define opacity here.
With Vue, property must be present on initial state if you want it to be 'reactive'. More info here : https://v1.vuejs.org/guide/reactivity.html

Comment (Ldoppea) [reply examples/components/RadarDemo.vue]: As you can see, with current code, visual opacity is not affected by code because reactivity is not set on that property.

File: examples/components/Radar.vue [hunk 13]
```diff
@@ -26,34 +37,83 @@ export default {
...
+    'config.direction': function() {
+      this.handleAngleChange()
+    },
+    'config.x': function() {
+      this.resetHandler()
     },
-    'config.direction': function (newVal, oldVal) {
-      this.handleDirectionChange(false);
+    'config.y': function() {
+      this.resetHandler()
```
Comment (Ldoppea) [inline examples/components/Radar.vue]: maybe `resetHandler()` can be done directly on `onDragEnd()`?

File: examples/components/Radar.vue (hunk 11 above)
Comment (imudin) [reply examples/components/Radar.vue]: yes I cannot animate radar and its handler in the same time :)

File: examples/components/RadarDemo.vue (hunk 12 above)
Comment (imudin) [reply examples/components/RadarDemo.vue]: Yeah it is not affected, let me try again

File: examples/components/Radar.vue (hunk 13 above)
Comment (imudin) [reply examples/components/Radar.vue]: yes right, now x and y no need be watched

**Comment by rafaesc** [comment]
Perfect! @imudin, @Ldoppea.

Thanks for your PR, I have already updated the readme and the websites with your contributions

https://github.com/rafaesc/vue-konva/commit/e2357b74e0125d0edb616305b9c1d4b6d301861b

Regards!
